app/frontend/src/features/profile/profilePage/ProfilePage.tsx

```diff
@@ -43,47 +42,60 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-type DetailType = keyof typeof SECTION_LABELS;
-
 export default function ProfilePage() {
   const classes = useStyles();
-  const { data: user } = useCurrentUser();
-  const history = useHistory();
-  const { type = "about" } = useParams<{ type: DetailType }>();
-  const detailType = type in SECTION_LABELS ? (type as DetailType) : "about";
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+  const [user, setUser] = useState<User.AsObject | null>(null);
+  const [currentTab, setCurrentTab] = useState<keyof typeof SECTION_LABELS>("about");
+
+  const { username } = useParams<{
+    username?: string;
+  }>();
+
+  useEffect(() => {
+    (async () => {
+        setLoading(true);
+        try {
+          setUser(await (username ? service.user.getUser(username) : getCurrentUser()));
+        } catch (e) {
+          console.error(e);
+          setError(e.message);
+        }
+        setLoading(false);
+    })();
+  }, [username]);
```

Comment:
Hmm it'd be more ideal to use the existing `useCurrentUser` or `useUserByUsername` hook instead since it uses React Query so we get these error and loading state for free from the lib. Also it re-uses users that have been previously fetched from the cache while it refetches in the background so things will be faster to load too.

Although not sure we can do that with how it's currently structured since we can't conditionally use a hook in React...

Reply:
The best way I can think of is to store the username in the auth state, so we can default to the current user's username if one isn't given by the URL. Thoughts on this @lucaslcode ?